Added Compositions for Western Sahara DLC

Cynicus-Nihilisticus opened the pull request:
Hi Tinter, I've created a full set of compositions for the Western Sahara DLC buildings. This adds support for all the new houses and includes the necessary changes to buildings.hpp and import.sqf. Hope this is helpful!

Cynicus-Nihilisticus commented:
Thanks for the detailed feedback, I really appreciate it.

I'm still new to this side of scripting, so I added the translation layer because I saw it used elsewhere and thought it was the standard procedure. I didn't fully understand the inheritance logic.

You are absolutely right that my changes will break the WS_lived_in_CUP.sqf and WS_lived_in_Cup_alt.sqf files. I tested it on my side, and the only way I could get those one's to work again was by updating their names to the translated names.

 I can easily update my compositions to use the full classnames as you suggested and remove the translation layer to fix this.

Before I do that, I'll run the test you asked for.

Tinter commented:
Sounds great, I know these details aren't really documented, so no fault for not realising how they work. I should add some documentation about how classnames resolve.
I've had a look ingame, I can see the names translated to are not actually classnames at all. The out.sqf will use the translation layer, so when you exported your compositions, it used the translated classnames. So the translation should be removed and your compositions should be adjusted to use the actual classnames for the buildings. 

I've had a look ingame and I've noticed some of the compositions place stuff outside, resulting in some furniture clipping into other objects or floating. Two examples are [5694.79,5097.8,0.00159454] and [5689.5,5127.47,0.00288773]. This is the first composition of "Land_House_C_5_EP1_off_lxWS" and "Land_House_K_1_EP1_lxWS" respectively, but I haven't checked every house, so I can't rule out there aren't more. I would discourage placing stuff outside as you just can't trust that houses will be placed at ground level or won't have their own objects around them.

Those issues aside, I really like the compositions. I was just going around looking inside the houses, and I really liked how subtle some of them were, I had to turn off the furniture and see it completely empty to make sure it wasn't just like that in vanilla. Great job, I really want to add them when the issues are fixed and then push an update to the mod.

Cynicus-Nihilisticus commented:
Ok, so I've gone through and made all the necessary changes.

- I've updated the `WS.sqf` file to use the correct, full classnames for the buildings.
- I've completely removed the translation namespaces from `buildings.hpp` as you suggested.
- I've also gone through the compositions and removed all the exterior items that were at risk of clipping or floating. Everything should be contained within the building structures now.

I'm really glad you like the compositions. I also made sure to use ZEI's "mark building positions" tool to see the AI pathing inside the houses and did my best to keep furniture out of their way.

I've tested the new files, and everything seems to be working 100% now. It also runs alongside `WS_lived_in_Cup.sqf` and `WS_lived_in_Cup_alt.sqf` without any issues.

Let me know if there's anything else needed!
